test/Microsoft.DocAsCode.Tests.Common/TestLoggerListener.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Tests.Common
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Microsoft.DocAsCode.Common;
+
+    public class TestLoggerListener : ILoggerListener
+    {
+        public List<ILogItem> Items { get; } = new List<ILogItem>();
+
+        private Func<ILogItem, bool> _matcher;
+
+        private TestLoggerListener(Func<ILogItem, bool> matcher)
+        {
+            _matcher = matcher;
+        }
+
+        #region ILoggerListener
+
+        public void WriteLine(ILogItem item)
+        {
+            if (item == null)
+            {
+                throw new ArgumentNullException(nameof(item));
+            }
+            if (_matcher(item))
+            {
+                Items.Add(item);
+            }
+        }
+
+        public void Flush()
+        {
+        }
+
+        public void Dispose()
+        {
+        }
+
+        #endregion
+
+        #region Creators
+
+        public static TestLoggerListener CreateLoggerListener(Func<ILogItem, bool> matcher)
+        {
+            return new TestLoggerListener(matcher);
+        }
+
+        public static TestLoggerListener CreateLoggerListenerWithPhaseStartMatcher(string phase, LogLevel logLevel = LogLevel.Warning)
+        {
+            return new TestLoggerListener(iLogItem =>
+            {
+                if (iLogItem.LogLevel < logLevel)
+                {
+                    return false;
+                }
+                if (phase == null ||
+                   (iLogItem?.Phase != null && iLogItem.Phase.StartsWith(phase, StringComparison.OrdinalIgnoreCase)))
```

Comment:
discuss offline, the comparision should be case sensitive.

---
In reply to: [119785752](https://github.com/dotnet/docfx/pull/1712#discussion_r119785752) [](ancestors = 119785752)